dbms/src/Flash/Mpp/MPPTask.cpp

```diff
@@ -405,10 +405,18 @@ void MPPTask::runImpl()
         auto result = query_executor_holder->execute();
         if (likely(result.is_success))
         {
-            // finish receiver
-            receiver_set->close();
+            /// Need to finish writing before closing the receiver.
+            /// For example, for the limit. Finishing writing first ensures that the limit on the TiDB side can end normally,
```

Comment:
```suggestion
            /// For example, for the limit executor. Finishing writing first ensures that the limit on the TiDB side can end normally,
```

Reply:
ok, refine the comment as
```
             /// For example, for the query with limit, calling `finishWrite` first to ensure that the limit executor on the TiDB side can end normally,
            /// otherwise the upstream MPPTasks will fail because of the closed receiver and then passing the error to TiDB.
```